src/Http/Controllers/Web/CategoryController.php

```diff
@@ -62,7 +63,27 @@ public function show(Request $request, Category $category): View
             ->orderBy('updated_at', 'desc')
             ->paginate();
 
-        return ViewFactory::make('forum::category.show', compact('privateAncestor', 'categories', 'category', 'threads'));
+        $selectableThreadIds = [];
+        if ($request->user()) {
+            if (Gate::any(['moveThreadsFrom', 'lockThreads', 'pinThreads'], $category)) {
+                // There are no thread-specific abilities corresponding to these,
+                // so we can include all of the threads for this page
+                $selectableThreadIds = $threads->pluck('id')->toArray();
+            } else {
+                $canDeleteThreads = $request->user()->can('deleteThreads', $category);
+                $canRestoreThreads = $request->user()->can('restoreThreads', $category);
```

Comment:
If both of these are false you could break out early?

Reply:
Good catch, I'll do that.